ee/fe/modules/500-basic-auth/templates/deployment.yaml

```diff
@@ -23,6 +23,7 @@ metadata:
   {{- include "helm_lib_module_labels" (list . (dict "app" .Chart.Name)) | nindent 2 }}
 spec:
   {{- include "helm_lib_deployment_strategy_and_replicas_for_ha" . | nindent 2 }}
+  revisionHistoryLimit: 2
```

Comment:
I see we have 2 as default. I would write a helm line for that, so "2" could be defined only once. And when it changes again, it could produce less diff. With an optional argument, 2 could be overwritten where needed.

Reply:
good point, i think we need helm lib helper for this

Reply:
Ok, I can rewrite the code using a helper, but I will explain my thoughts first. 

Key points:
1. If we define the helper, it should be used everywhere in the code to keep the same style. It is terrible when we use helpers somewhere and somewhere not. For example:
```
{{ include "helm_lib_revision_history_limit" . }} # revisionHistoryLimit: 2
```
```
revisionHistoryLimit: 0
```

2. To write the helper with the ability to override the default, we still need to use a value. For example:
```
{{ include "helm_lib_revision_history_limit" . }} # revisionHistoryLimit: 2
```
```
{{ include "helm_lib_revision_history_limit" 0 }} # revisionHistoryLimit: 0
```
3. Splitting into two different helpers is also possible but seems like an overkill
```
{{ include "helm_lib_default_revision_history_limit" . }} # revisionHistoryLimit: 2
```
```
{{ include "helm_lib_revision_history_limit" 0 }} # revisionHistoryLimit: 0
```
4. Current variant.
```
revisionHistoryLimit: 2
```
```
revisionHistoryLimit: 0
```
I picked the last one because, for me, it looks less wordy and explicit enough.

Reply:
Which one should I pick?

Reply:
How likely this "2" is gonna change to e.g. 1 or 3?

Reply:
Probably my intention to save the diff is just not that important in case of 'find-and-replace one line' scenario. Templating one line looks like an overcomplication for no real value. Thanks for describing all cases. 

Do we want linter for this field to ensure nobody forgets to set 2 or less?